pkg/webhook/workspace/handler/immutable.go

```diff
@@ -81,8 +84,26 @@ func (h *WebhookHandler) handleImmutableRoute(oldObj, newObj runtime.Object, use
 	if username == h.ControllerSAName {
 		return true, ""
 	}
-	if reflect.DeepEqual(oldObj, newObj) {
+	if changePermitted(oldObj, newObj) {
 		return true, ""
 	}
 	return false, "object is owned by workspace and cannot be updated."
 }
+
+func changePermitted(oldObj, newObj runtime.Object) bool {
+	oldCopy := oldObj.DeepCopyObject()
+	newCopy := newObj.DeepCopyObject()
+	oldMeta, ok := oldCopy.(metav1.Object)
+	if !ok {
+		log.Error(fmt.Errorf("Object is not a valid k8s object: does not have metadata"), "Failed to compare objects")
+		return false
+	}
+	newMeta, ok := newCopy.(metav1.Object)
+	if !ok {
+		log.Error(fmt.Errorf("Object is not a valid k8s object: does not have metadata"), "Failed to compare objects")
+		return false
+	}
+	newMeta.SetFinalizers(oldMeta.GetFinalizers())
+	newMeta.SetDeletionTimestamp(oldMeta.GetDeletionTimestamp())
+	return reflect.DeepEqual(oldMeta, newMeta)
```

Comment:
Can't we compare it in the same way as we do for workspace? I assume there is trouble with types but just reading)

Reply:
The main issue is that [`cmp` is meant for tests only](https://godoc.org/github.com/google/go-cmp/cmp):

> It is intended to only be used in tests, as performance is not a goal and it may panic if it cannot compare the values. Its propensity towards panicking means that its unsuitable for production environments where a spurious panic may be fatal. 

so we probably shouldn't be using it at all :)